src/cohortManager.js

```diff
@@ -0,0 +1,189 @@
+import Cohort from './cohort.js'
+import Student from './student.js'
+
+class CohortManager {
+  constructor() {
+    this.cohorts = []
+    this.id = 1
+  }
+
+  validateCohortName(cohortName) {
+    if (typeof cohortName !== 'string') {
+      throw new Error("Please input a valid name e.g. 'cohort-11'")
+    }
+  }
+
+  validateStudent(student) {
+    if (
+      !student.firstName ||
+      !student.lastName ||
+      !student.github ||
+      !student.email
+    ) {
+      throw new Error(
+        "Please input a valid student e.g. {firstName: 'John', lastName: 'Doe', github: '@johndoe', email:'johndoe@mail.com'}"
+      )
+    }
+  }
+
+  createCohort(cohortName) {
+    this.validateCohortName(cohortName)
+
+    const duplicate = this.cohorts.some((cohort) => cohort.name === cohortName)
+
+    if (!duplicate) {
+      const cohort = new Cohort(cohortName)
+      this.cohorts.push(cohort)
+
+      return this.cohorts.find((cohort) => cohort.name === cohortName)
+    } else {
+      return `${cohortName} already exists`
+    }
+  }
+
+  findCohortByName(cohortName) {
+    this.validateCohortName(cohortName)
+
+    const hasCohort = this.cohorts.some((cohort) => cohort.name === cohortName)
+
+    if (hasCohort) {
+      return this.cohorts.find((cohort) => cohort.name === cohortName)
+    } else {
+      throw new Error('Cohort does not exist')
+    }
```

Comment:
You don't need the else clause in here, you can just throw the error by default at the bottom of the function